src/mqt/ddsim/qasmsimulator.py

```diff
@@ -73,22 +76,70 @@ def target(self):
     def max_circuits(self):
         return None
 
-    def run(self, quantum_circuits: QuantumCircuit | list[QuantumCircuit], **options: dict[str, Any]) -> DDSIMJob:
+    @staticmethod
+    def _bind_parameters(
+        quantum_circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]] | Sequence[Mapping[Parameter, float]] | None,
+    ) -> list[QuantumCircuit]:
+        if parameter_values is None:
+            parameter_values = []
+
+        number_parametrized_circuits = 0  # Initialize the counter
+
+        for qc in quantum_circuits:
+            if qc.parameters:
+                number_parametrized_circuits += 1
+
+        if number_parametrized_circuits != 0 or len(parameter_values) != 0:
+            if number_parametrized_circuits == 0:
+                msg = "No parametrized circuits found in the provided list. The parameter list should be of type None or an empty list."
+                raise ValueError(msg)
+            if len(parameter_values) != len(quantum_circuits):
+                msg = "The number of circuits to simulate does not match the size of the parameter list."
+                raise ValueError(msg)
+            bound_circuits = []
+            for qc, values in zip(quantum_circuits, parameter_values):
+                if len(qc.parameters) != len(values):
+                    msg = f"The number of parameters in the circuit '{qc.name}' does not match the number of parameters provided. Expected number of parameters is '{len(qc.parameters)}'."
```

Comment:
```suggestion
                    msg = f"The number of parameters in the circuit '{qc.name}' does not match the number of parameters provided ({len(values)}). Expected number of parameters is '{len(qc.parameters)}'."
```
Include slightly more information